### PR TITLE
chore(dev): upgrade docker base image

### PR DESCRIPTION
**Pull request by outsinre**
### Summary

Upgrade docker base image.

### Checklist

- [n/a] The Pull Request has tests
- [n/a] There's an entry in the CHANGELOG
- [n/a] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* upgrade docker base image (rpm, apk, and deb)

### Issue reference

Fix #FTI-5195

### CODE REVIEW

File: build/dockerfiles/deb.Dockerfile [hunk 1]
```diff
@@ -18,6 +18,7 @@ ARG KONG_ARTIFACT_PATH=
 COPY ${KONG_ARTIFACT_PATH}${KONG_ARTIFACT} /tmp/kong.deb
 
 RUN apt-get update \
+    && apt-get -y upgrade \
```
Comment (dndx) [inline build/dockerfiles/deb.Dockerfile:21]: Maybe also add `apt autoremove` afterwards to remove unused dependencies?

Comment (outsinre) [reply build/dockerfiles/deb.Dockerfile:21]: Make sense. Updated.

**Review by dndx — APPROVED** [review]

**Comment by dndx** [comment]
@outsinre This one should be backported to 2.8 and 3.x as it impacts existing user.

**Comment by outsinre** [comment]
Will track the backport in FTI tickets.

**Comment by fffonion** [comment]
note we don't do backports in CE

**Pull request merged**